### PR TITLE
Delete code for obsolete Harmony support

### PR DESCRIPTION
**Pull request by pshipton**
Includes removing the j9vmi library. 
Also OPT_PASS which is unused and depends on OPT_HARMONY.

### CODE REVIEW

File: runtime/include/j9memcategories.h [hunk 1]
```diff
@@ -52,7 +52,7 @@
 #define J9MEM_CATEGORY_UNUSED10 10 /* #define J9MEM_CATEGORY_JIT 10 */
 #define J9MEM_CATEGORY_UNUSED11 11 /* #define J9MEM_CATEGORY_JIT_CODE_CACHE 11 */
 #define J9MEM_CATEGORY_UNUSED12 12 /* #define J9MEM_CATEGORY_JIT_DATA_CACHE 12 */
-#define J9MEM_CATEGORY_HARMONY 13
+#define J9MEM_CATEGORY_UNUSED13 13 /* #define J9MEM_CATEGORY_HARMONY 13 */
```
Comment (keithc-ca) [inline runtime/include/j9memcategories.h]: I don't think there's much value in keeping this comment.

**Comment by keithc-ca** [comment]
The travis build failed:
```
[ 34%] Linking CXX shared library ../libjvm.so
/usr/bin/ld: cannot find -lj9vmi
```
It seems there's at least one reference to `j9vmi` left to be removed.

File: test/functional/Java8andUp/src/j9vm/test/memory/MemoryAllocator.java [hunk 2]
```diff
@@ -334,7 +333,7 @@ protected boolean isJavacore(File f) {
 		// categoryCodesToNames.put(10l, "JIT"); //#define J9MEM_CATEGORY_JIT 10 - Moved to omr
 		// categoryCodesToNames.put(11l, "JIT Code Cache"); //#define J9MEM_CATEGORY_JIT_CODE_CACHE 11 - Moved to omr
 		// categoryCodesToNames.put(12l, "JIT Data Cache"); //#define J9MEM_CATEGORY_JIT_DATA_CACHE 12 - Moved to omr
-		categoryCodesToNames.put(13l, "Harmony Class Libraries"); //#define J9MEM_CATEGORY_HARMONY 13
+		// categoryCodesToNames.put(13l, "Harmony Class Libraries"); //#define J9MEM_CATEGORY_HARMONY 13
```
[the remark this answers is not in the document]
Comment (pshipton) [reply test/functional/Java8andUp/src/j9vm/test/memory/MemoryAllocator.java]: I'm following the pattern of the previous categories removed.

File: runtime/include/j9memcategories.h (hunk 1 above)
Comment (pshipton) [reply runtime/include/j9memcategories.h]: I'm following the pattern of the previous categories removed.

**Comment by pshipton** [comment]
Yes, need to figure out if it's really safe the remove j9vmi, and then there any many references to remove.

**Comment by keithc-ca** [comment]
I expect it is safe to remove `j9vmi`: it's an object library which would be empty if kept.

File: runtime/include/j9memcategories.h (hunk 1 above)
Comment (keithc-ca) [reply runtime/include/j9memcategories.h]: Got it. I don't expect we need to keep the values the same from release to release.

**Comment by pshipton** [comment]
> I expect it is safe to remove j9vmi: it's an object library which would be empty if kept.

Well yes, it either needs to be removed, or I need to put the guts of it back in. I think it's safe to remove, googling didn't find any useful references to the APIs.

**Comment by pshipton** [comment]
I will squash this before merge, but keeping a second commit while testing in case j9vmi needs to be restored.

**Comment by keithc-ca** [comment]
Jenkins test sanity win jdk8

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk11

**Comment by pshipton** [comment]
The PR testing passed, although the github status didn't get updated. I'm going to add another commit to remove the OPT_HARMONY build flag, which is also obsolete.
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2750/
https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/2751/

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk11

**Pull request merged**